### PR TITLE
Feature/currency v2

### PR DESCRIPTION
**Pull request by mvenaas**
The currency module is now alive!

### CODE REVIEW

File: BuffBot/database.py [hunk 1]
```diff
@@ -49,35 +49,42 @@ def get_game_channel(self, title):
...
         self.conn.close()
 
+    @classmethod
     def get_coin_count(self, user_id):
         # TODO: Implement total coin value in DB as separate row
-        total_coins = 0
-        self.conn = sqlite3.connect(self.DB_NAME)
-        sessions = self.conn.execute("select start_time, end_time from coins where user_id = {} and session_active = 0"
-                                  .format(user_id))
+        self.conn = sqlite3.connect("test123.db")
```
Comment (Cmoen11) [inline BuffBot/database.py]: Please edit this to self.DB_NAME so it's easier to change db name in the future

File: BuffBot/database.py [hunk 2]
```diff
@@ -49,35 +49,42 @@ def get_game_channel(self, title):
...
+        self.conn.execute(sql, params)
         self.conn.commit()
         self.conn.close()
 
     @classmethod
     def set_coin_count_session_end(self, user_id, end_time, session_active):
-        self.conn = sqlite3.connect(self.DB_NAME)
-        sql = 'UPDATE coins SET (end_time = {}, session_active = {}) ' \
-              'WHERE user_id = "{}"'.format(end_time, session_active, user_id)
+        self.conn = sqlite3.connect("test123.db")
```
Comment (Cmoen11) [inline BuffBot/database.py]: Please edit this to self.DB_NAME so it's easier to change db name in the future

File: BuffBot/database.py [hunk 3]
```diff
@@ -49,35 +49,42 @@ def get_game_channel(self, title):
...
-        sessions = self.conn.execute("select start_time, end_time from coins where user_id = {} and session_active = 0"
-                                  .format(user_id))
+        self.conn = sqlite3.connect("test123.db")
+        params = (user_id, 0)
+        sql = "SELECT start_time, end_time FROM coins WHERE user_id =? and session_active = ?"
+
+        sessions = self.conn.execute(sql, params)
 
+        def format_2_dec(val):
+            return "{:.2f}".format(val)
```
Comment (Cmoen11) [inline BuffBot/database.py]: What is this? Please comment 👍

File: BuffBot/currency.py [hunk 4]
```diff
@@ -1,27 +1,33 @@
...
 
-    # TODO Make a session persist betweeen voice room changes if not in AFK
+    # TODO Make a session persist between voice room changes if not in AFK
     def register_activity(self, before, after):
 
         if check_if_connected(after) and not member_in_channel("AFK", after):
             print("You're in", after.name)
-            # TODO These database calls does not currently work
-            # database.Database.set_coin_count_session_start(after.id, time.gmtime(), 0, 1)
+            print(time.time())
```
Comment (Cmoen11) [inline BuffBot/currency.py:14]: I think we should start removing prints in our code.. Please keep just prints who is to help to the end user.

**Review by Cmoen11 — CHANGES_REQUESTED** [review]
Please edit by feedback given

File: BuffBot/currency.py [hunk 5]
```diff
@@ -1,27 +1,33 @@
...
+            database.Database.set_coin_count_session_start(after.id, time_now(), 0, 1)
         elif not member_in_channel("AFK", after):
             print("You're out", after.name)
-            # database.Database.set_coin_count_session_end(before.id, time.gmtime(), 0)
+            database.Database.set_coin_count_session_end(before.id, time_now(), 0)
 
         # If user is in a channel named AFK, end session. IE: not get coins for being afk
         if member_in_channel("AFK", after):
             print("You're afk")
             # database.Database.set_coin_count_session_end(before.id, time.gmtime(), 0)
```
Comment (martinnenseth) [inline BuffBot/currency.py:23]: This if statement should do something when it triggers.

Comment (Cmoen11) [reply BuffBot/currency.py:23]: Instead of printing stuff everytime someone is afk, is it just better to remove it?

**Review by martinnenseth — CHANGES_REQUESTED** [review]
Good job! Make the necessary small changes, and this is definitly approved!

File: BuffBot/currency.py (hunk 5 above)
Comment (martinnenseth) [reply BuffBot/currency.py:23]: Seems like I targeted the wrong area of the code... 
I was thinking about line 21.     #If user is in a channel named AFK, end session. IE: not get coins for being afk
This logic should be implemented

Comment (Cmoen11) [reply BuffBot/currency.py:23]: Yeah.. But it'll only give coins `elif not member_in_channel("AFK", after):` 
On line 18: `database.Database.set_coin_count_session_end(before.id, time_now(), 0)` therefor line 21 and down is not nessecary

**Comment by Cmoen11** [comment]
Corrected some of the requirements for the pullrequest.. And now able to merge this in

**Pull request merged**